dowhy/causal_refuters/partial_linear_sensitivity_analyzer.py

```diff
@@ -0,0 +1,457 @@
+            probability = scipy.stats.norm.ppf(1 - significance_level)
+            lower_confidence_bound = theta_lower - probability * \
+                np.sqrt(np.mean(stddev_lower * stddev_lower) +
+                        np.var(theta_lower))
+            upper_confidence_bound = theta_upper + probability * \
+                np.sqrt(np.mean(stddev_upper * stddev_upper) +
+                        np.var(theta_upper))
+
+        else:
+            lower_confidence_bound = theta_lower
+            upper_confidence_bound = theta_upper
+
+        return lower_confidence_bound, upper_confidence_bound, bias
+
+    def calculate_robustness_value(self, alpha):
+        """
+                Function to compute the robustness value of estimate against the confounders
+                :param alpha: confidence interval for statistical inference
+
+                :returns: robustness value
+        """
+        for t_val in np.arange(0, 1, 0.01):
+            lower_confidence_bound, _, _ = self.get_confidence_levels(
+                r2yu_tw=t_val, r2tu_w=t_val, significance_level=alpha)
+            if lower_confidence_bound <= 0:
+                return t_val
+        return t_val
+
+    def perform_benchmarking(self, r2yu_tw, r2tu_w):
+        """
+        :param r2yu_tw: proportion of residual variance in the outcome explained by confounders
+        :param r2tu_w: proportion of residual variance in the treatment explained by confounders
+
+        :returns: python dictionary storing values of r2tu_w, r2yu_tw, short estimate, bias, lower_ate_bound,upper_ate_bound, lower_confidence_bound, upper_confidence_bound
+        """
+
+        lower_confidence_bound, upper_confidence_bound, bias = self.get_confidence_levels(
+            r2yu_tw=r2yu_tw, r2tu_w=r2tu_w, significance_level=0.05)
+        lower_ate_bound, upper_ate_bound, bias = self.get_confidence_levels(
+            r2yu_tw=r2yu_tw, r2tu_w=r2tu_w, significance_level=None)
+
+        benchmarking_results = {
+            'r2tu_w': r2tu_w,
+            'r2yu_tw': r2yu_tw,
+            'short estimate': self.theta_s,
+            'bias': bias,
+            'lower_ate_bound': lower_ate_bound,
+            'upper_ate_bound': upper_ate_bound,
+            'lower_confidence_bound': lower_confidence_bound,
+            'upper_confidence_bound': upper_confidence_bound
+        }
+
+        return benchmarking_results
+
+    def get_regression_partial_r2(self, X, Y, numeric_features, split_indices):
+        """
+        Calculates the pearson non parametric partial R^2 from a regression function.
+
+        :param X: numpy array containing set of regressors 
+        :param Y: outcome variable in regression
+        :param numeric_features: list of indices of columns with numeric features
+        :param split_indices: training and testing data indices obtained after cross folding
+
+        :returns: partial R^2 value
+        """
+        regression_model = get_generic_regressor(cv=split_indices,

```

Comment:
In case of partial linear, what are the allowed models for generic_regressor? Since this is first-stage for treatment or outcome, I guess non-linear models are also allowed. Is that correct?

Reply:
By default , we have RandomForestRegressor, Reisz Regressor and a lasso pipeline. Yes non-linear models are allowed.